dbt/include/athena/macros/materializations/seeds/helpers.sql

```diff
@@ -18,27 +18,113 @@
 
-  {% set sql %}
-    create external table {{ this.render_hive() }} (
+  {%- set s3_location = adapter.upload_seed_to_s3(
+    s3_data_dir,
+    s3_data_naming,
+    external_location,
+    model.schema,
+    model.name,
+    agate_table,
+  ) -%}
+
+  -- create tmp relation
+  {%- set tmp_relation = api.Relation.create(
+    identifier=model.name + "_tmp",
+    schema=model.schema,
+    database=model.database,
+    type='table'
+  ) -%}
+
+  -- create target relation
+  {%- set relation = api.Relation.create(
+    identifier=model.name,
+    schema=model.schema,
+    database=model.database,
+    type='table'
+  ) -%}
+
+  -- drop tmp relation if exists
+  {{ drop_relation(tmp_relation) }}
+
+  {% set sql_tmp_table %}
+    create external table {{ tmp_relation.render_hive() }} (
         {%- for col_name in agate_table.column_names -%}
-            {%- set inferred_type = adapter.convert_type(agate_table, loop.index0) -%}
-            {%- set type = column_override.get(col_name, inferred_type) -%}
             {%- set column_name = (col_name | string) -%}
-            {{ adapter.quote_seed_column(column_name, quote_seed_column) }} {{ type }} {%- if not loop.last -%}, {% endif -%}
+            {{ adapter.quote_seed_column(column_name, quote_seed_column) }} string {%- if not loop.last -%}, {% endif -%}
         {%- endfor -%}
     )
-    stored as parquet
-    location '{{ adapter.s3_table_location(s3_data_dir, s3_data_naming, model["schema"], model["alias"], external_location) }}'
-    tblproperties ('classification'='parquet')
+    row format serde 'org.apache.hadoop.hive.serde2.OpenCSVSerde'
+    location '{{ s3_location }}'
+    tblproperties (
+      'skip.header.line.count'='1'
+    )
   {% endset %}
 
+  -- casting to type string is not allowed needs to be varchar
+  {% set date_formats = [
+    '%Y-%m-%d %H:%i:%s',
+    '%Y/%m/%d %H:%i:%s',
+    '%d %M %Y %H:%i:%s',
+    '%d/%m/%Y %H:%i:%s',
+    '%d-%m-%Y %H:%i:%s',
+    '%Y-%m-%d %H:%i:%s.%f',
+    '%Y/%m/%d %H:%i:%s.%f',
+    '%d %M %Y %H:%i:%s.%f',
+    '%d/%m/%Y %H:%i:%s.%f',
+    '%Y-%m-%dT%H:%i:%s.%fZ',
+    '%Y-%m-%dT%H:%i:%sZ',
+    '%Y-%m-%dT%H:%i:%s',
+  ]%}
+
+  {% set sql %}
+    select
+        {% for col_name in agate_table.column_names -%}
+            {%- set inferred_type = adapter.convert_type(agate_table, loop.index0) -%}
+            {%- set type = column_override.get(col_name, inferred_type) -%}
+            {%- set type = type if type != "string" else "varchar" -%}
+            {%- set column_name = (col_name | string) -%}
+            {%- set quoted_column_name = adapter.quote_seed_column(column_name, quote_seed_column) -%}
+            {% if type == 'timestamp' %}
+            coalesce(
+              {% for date_format in date_formats %}
+                try(date_parse({{ quoted_column_name }}, '{{ date_format }}'))
+                {%- if not loop.last -%}, {% endif -%}
+              {% endfor %}
+            ) as {{quoted_column_name}}
```

Comment:
As a nitpick, we might want to move it to an other file as a macro ?

Reply:
@Jrmyy I could add it as a separate macro in the same file? Does that work for you?

Reply:
@henriblancke yes should be fine, as it's an "helper.sql"

Reply:
Yes that works 👍🏻